src/services/search-builder.js

```diff
@@ -164,18 +164,15 @@ function SearchBuilder(model, opts, params, fieldNamesRequested) {
               const column = opts.sequelize.col(`${association.as}.${columnName}`);
 
               if (field.field === schemaAssociation.idField) {
+                let value = params.search;
+
+                // NOTICE: cast to Number to see if positive number
                 if (field.type === 'Number') {
-                  const value = parseInt(params.search, 10) || 0;
-                  if (value) {
-                    condition = opts.sequelize.where(
-                      column,
-                      ' = ',
-                      parseInt(params.search, 10) || 0,
-                    );
-                    hasExtendedConditions = true;
-                  }
-                } else if (params.search.match(REGEX_UUID)) {
```

Comment:
I am pretty sure the query will crash if you build an equality condition on UUID column with a value that has not a UUID format, and thus introduce a regression here.

Reply:
Indeed!
Thanks for pointing this out!
I've been able to reproduce the error and to fix it :)